### ISSUE TITLE
Hyperion prematurely quits when stdin is closed in daemon mode

### ISSUE THREAD

**Issue by bluecmd** [SDL-Hercules-390/hyperion#162]
Hi,

I'm running Hyperion in daemon mode (just upgraded from Hercules 3.12).

I run it like this:
```
hercules -f /home/hercules/x.conf -r hercules.rc -d
```

However, when the rc file reaches its end Hyperion terminates. This is different from how Hercules used to behave and I think it might be a bug.

This is the log from the behavior:
```
...
Dec 22 11:49:57 z ip[10006]: HHC02263I Script 1: processing resumed...
Dec 22 11:49:57 z ip[10006]: HHC02264I Script 1: file hercules.rc processing ended
Dec 22 11:49:57 z ip[10006]: HHC02260I Script 2: begin processing file <stdin>
Dec 22 11:49:57 z ip[10006]: HHC02264I Script 2: file <stdin> processing ended
Dec 22 11:49:57 z ip[10006]: HHC00101I Thread id 00007f8653769700, prio -1, name 'Processor CP00' ended
Dec 22 11:49:57 z ip[10006]: HHC01420I Begin Hercules shutdown
```

It looks like there was an assumption that if stdin closes the user wants to terminate, but in daemon mode that doesn't make much sense I think.

**Comment by bluecmd** [comment]
Aha, looking at the code it seems like the behavior is to terminate only if there are no CPUs actively running.
Pretty nifty, but would probably be nice to print a message letting the user know that that is why the software terminates.

https://github.com/SDL-Hercules-390/hyperion/blob/c8addaafe1133427607280a46536844284f4e974/impl.c#L1247-L1254

**Comment by Fish-Git** [comment]
> This is what I patched with:

_Good catch, Christian! **Thank you!**_

I will get that fixed right away!

**Comment by Fish-Git** [comment]
Fixed by commit 64b1b3f14f381b2af45c1188e12a8f94b2605e8a. Closing.

**Comment by bluecmd** [comment]
Thanks for the speedy fix :-) I wish you good holidays.
